packages/models-library/src/models_library/basic_types.py

```diff
@@ -80,6 +80,22 @@ class IDStr(ConstrainedStr):
     max_length = 100
 
 
+# Truncated Str:
+#   - Strips whitespaces and truncate strings that exceed the specified characters limit (curtail_length).
+#   - This ensures the **input** to the API is controlled and prevents exceeding the database's constraints
+#   silently, i.e. without raising errors.
+class ShortTruncatedStr(ConstrainedStr):
```

Comment:
I get the concept, but 1000 characters for a description is rather short right?
1. what is the limit in the database?
2. what is the rationale behind the 1000?
3. I would put these comments inside the classes so that they are kept together instead of outside.

Reply:
I have an additional point here: If this is to be used for the API server, it means that these will never get read by any human? In that case will it not make more sense to trim them down to length 0?

Reply:
Agree with the 1000 characters limit, this will eventually truncate the existing descriptions. Did you check how long are those in production?

Reply:
1. No, there is not limit in the database. I will rephrase the comment
2. Just took a text with a reasonable length and measured it. I reviewed some of our database and these are the maximum descriptions
```sql
SELECT LENGTH(description), description AS length
FROM services_meta_data
WHERE description IS NOT NULL
ORDER BY LENGTH(description) DESC
LIMIT 2
```
![image](https://github.com/ITISFoundation/osparc-simcore/assets/32402063/1bc79ab6-37fc-40af-aef0-aa255d5849d8)
- service_meta_data -> 97
- projects -> 1821

So I will update this to 2500 character

Reply:
> I have an additional point here: If this is to be used for the API server, it means that these will never get read by any human? In that case will it not make more sense to trim them down to length 0?

@GitHK  I am not sure I understand the rationale to trim them to zero. In any case, this entrypoint is not used by the api-server. Perhaps you can clarify it to me offline?

Reply:
I've read it wrong unfortunately. I see this is for the projects visible to the user.